### PR TITLE
[SPARK-20889][SparkR] Grouped documentation for COLLECTION column methods

### PR DESCRIPTION
**Pull request by actuaryzhang**
## What changes were proposed in this pull request?

Grouped documentation for column collection methods.

### CODE REVIEW

**Comment by actuaryzhang** [comment]
Last part of this doc update.

**Comment by actuaryzhang** [comment]
@felixcheung @HyukjinKwon 

image
image
image

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -2156,28 +2178,23 @@ setMethod("date_format", signature(y = "Column", x = "character"),
...
 #' @examples
+#'
 #' \dontrun{
-#' schema <- structType(structField("name", "string"),
-#' select(df, from_json(df$value, schema, dateFormat = "dd/MM/yyyy"))
-#'}
+#' df2 <- sql("SELECT named_struct('name', 'Bob') as people")
+#' df2 <- mutate(df2, people_json = to_json(df2$people))
+#' schema <- structType(structField("name", "string"))
+#' head(select(df2, from_json(df2$people_json, schema)))}
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: I think it's worthwhile to keep `dateFormat = "dd/MM/yyyy")` in the example

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -2156,28 +2178,23 @@ setMethod("date_format", signature(y = "Column", x = "character"),
...
 
-#' from_json
-#'
-#' Parses a column containing a JSON string into a Column of \code{structType} with the specified
-#' \code{schema} or array of \code{structType} if \code{as.json.array} is set to \code{TRUE}.
-#' If the string is unparseable, the Column will contains the value NA.
+#' @details
+#' \code{from_json}: Parses a column containing a JSON string into a Column of \code{structType}
+#' with the specified \code{schema} or array of \code{structType} if \code{as.json.array} is set
+#' to \code{TRUE}. If the string is unparseable, the Column will contains the value NA.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: btw, `will contains the value NA.` is very consistently documented. in this case this is right, but there are many other that says the value is `null` (note lower case) which isn't quite correct on the R side.

another project? :)

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -132,6 +132,35 @@ NULL
...
+#' # Dataframe used throughout this doc
+#' df <- createDataFrame(cbind(model = rownames(mtcars), mtcars))
+#' df <- createDataFrame(cbind(model = rownames(mtcars), mtcars))
+#' tmp <- mutate(df, v1 = create_array(df$mpg, df$cyl, df$hp))
+#' head(select(tmp, array_contains(tmp$v1, 21), size(tmp$v1)))
+#' tmp2 <- mutate(tmp, v2 = explode(tmp$v1))
+#' head(tmp2)
+#' head(select(tmp, posexplode(tmp$v1)))
+#' head(select(tmp, sort_array(tmp$v1)))
+#' head(select(tmp, sort_array(tmp$v1, FALSE)))}
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: nit, let's improve this? I think in sort_array we could be more clear, eg. `sort_array(tmp$v1, asc = FALSE)`

**Review by felixcheung — COMMENTED** [review]
LGTM - a few very minor comments

File: R/pkg/R/functions.R (hunk 2 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: Corrected the typo. Will consider updating `null` & `NA` in the future :)

File: R/pkg/R/functions.R (hunk 1 above)
Comment (actuaryzhang) [inline R/pkg/R/functions.R]: Thanks for catching this. Added an example.

**Comment by actuaryzhang** [comment]
@felixcheung This is the last set of this doc update. Once it gets in, I will do another pass to fix any styles or consistency issue.

**Review by felixcheung — APPROVED** [review]

**Comment by felixcheung** [comment]
merged to master. thanks!

**Pull request closed without merging**